batch/snippets/src/main/java/com/example/batch/CreateBatchCustomNetwork.java

```diff
@@ -0,0 +1,148 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.example.batch;
+
+// [START batch_create_custom_network]
+
+import com.google.cloud.batch.v1.AllocationPolicy;
+import com.google.cloud.batch.v1.BatchServiceClient;
+import com.google.cloud.batch.v1.CreateJobRequest;
+import com.google.cloud.batch.v1.Job;
+import com.google.cloud.batch.v1.LogsPolicy;
+import com.google.cloud.batch.v1.LogsPolicy.Destination;
+import com.google.cloud.batch.v1.Runnable;
+import com.google.cloud.batch.v1.Runnable.Script;
+import com.google.cloud.batch.v1.TaskGroup;
+import com.google.cloud.batch.v1.TaskSpec;
+import com.google.protobuf.Duration;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class CreateBatchCustomNetwork {
+
+  public static void main(String[] args)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    // TODO(developer): Replace these variables before running the sample.
+    // Project ID or project number of the Google Cloud project you want to use.
+    String projectId = "YOUR_PROJECT_ID";
+    // Name of the region you want to use to run the job. Regions that are
+    // available for Batch are listed on: https://cloud.google.com/batch/docs/get-started#locations
+    String region = "europe-central2";
+    // The name of the job that will be created.
+    // It needs to be unique for each project and region pair.
+    String jobName = "JOB_NAME";
+    // The name of a VPC network in the current project or a Shared VPC network that is hosted by
+    // or shared with the current project.
+    String network = "global/networks/test-network";
+    // The name of a subnet that is part of the VPC network and is located
+    // in the same region as the VMs for the job.
+    String subnet = "regions/europe-west1/subnetworks/subnet";
```

Comment:
Could you use String.format to denote which components of these variables need to be replaced?